src/app/pages/main/components/community-editor/community-editor.component.ts

```diff
@@ -0,0 +1,129 @@
+        lng: ['', Validators.required],
+      }),
+      description: ['', Validators.required], // ADD LENGTH VALIDATOR WITH COUNTER AT THE BOTTOM
+      city: ['', Validators.required],
+      twitter: ['', Validators.required],
+      web: [''],
+      youtube: [''],
+      organizers: formBuilder.array([this.createOrganizerFromGroup()]),
+    });
+  }
+
+  ngAfterViewInit(): void {
+    this.getPlaceAutocomplete();
+  }
+
+  private getPlaceAutocomplete() {
+    const autocomplete = new google.maps.places.Autocomplete(this.addresstext.nativeElement, {
+      types: ['address'], // 'establishment' / 'address' / 'geocode'
+    });
+    google.maps.event.addListener(autocomplete, 'place_changed', () => {
+      const place = autocomplete.getPlace();
+      const positionControls = this.communityForm['controls'].position['controls'];
+      positionControls.lat.setValue(place.geometry.location.lat());
+      positionControls.lng.setValue(place.geometry.location.lng());
+      this.autocompletedInput = place.name;
+    });
+  }
+
+  createOrganizerFromGroup() {
+    return this.formBuilder.group({
+      name: ['', Validators.required],
+      twitter: ['', Validators.required],
+      github: [''],
+      webs: this.formBuilder.array([this.formBuilder.control('')]),
+    });
+  }
+  addOrganizer() {
+    (this.communityForm.get('organizers') as FormArray).push(this.createOrganizerFromGroup());
+  }
+  addWeb(organizerIndex: number) {
+    (this.communityForm.get('organizers')['controls'][organizerIndex]['controls'][
+      'webs'
+    ] as FormArray).push(this.formBuilder.control(''));
+  }
+  onSubmit() {
+    let cleanedFormResult = this.cleanFormResult(this.communityForm.value);
+    let copyOperation = this._clipboard.beginCopy(
+      `"${cleanedFormResult.name}": ` + JSON.stringify(cleanedFormResult),
+    );
+    copyOperation.copy();
+    copyOperation.destroy();
+
+    this._snackBar.open('Community JSON copied into clipboard', 'OK', {
+      duration: 2000,
+      panelClass: 'custom-mat-snackbar-style',
+    });
+  }
+
+  cleanFormResult(formResult: any) {
+    // This functions should be tested
+    let cleanedResult = { ...formResult };
+    cleanedResult['name'] = this.autocompletedInput;
+    cleanedResult['id'] = this.lastCommunityId + 1;
+
+    // Remove empty optional inputs
+    cleanedResult.organizers.forEach(organizer => {
+      if (organizer.github === '') delete organizer.github;
+      if (organizer.webs && organizer.webs.length === 0) delete organizer.webs;
+      else if (!organizer.webs) {
+        delete organizer.webs;
+      } else {
+        organizer.webs = organizer.webs.filter(web => web != '');
+        if (organizer.webs.length === 0) delete organizer.webs;
+      }
+    });
+    delete cleanedResult.addressName;
+    if (cleanedResult.web === '') delete cleanedResult.web;
+    if (cleanedResult.youtube === '') delete cleanedResult.youtube;
+
+    return cleanedResult;
```

Comment:
```suggestion
    const {
    	addressName: _,
    	organizers,
    	web,
    	youtube,
    	...community,
    } = this.communityForm.value;
   
    return {
     ...community,
     name: this.autocompletedInput,
     id: this.communities.lenght,
     web: web || undefined,
     youtube: youtube || undefined,
     organizers: organizers.map(
     	({ github, webs, ...organizer }) => ({
    		 ...organizer,
     		github: github || undefiend,
     		webs: Array.isArray(webs) && webs.length ? webs.filter(Boolean) : undefined,
     	}),
     };
```

Reply:
There is a typo in:
`id: this.communities.lenght`

It would be ".length"

Reply:
Good catch! I have edited it :smile: